packages/turbo-workspaces/src/managers/npm.ts

```diff
@@ -71,39 +73,57 @@ async function read(args: ReadArgs): Promise<Project> {
  */
 async function create(args: CreateArgs): Promise<void> {
   const { project, options, to, logger } = args;
+  const hasWorkspaces = project.workspaceData.globs.length > 0;
 
-  logger.mainStep(`Creating npm workspaces`);
+  logger.mainStep(
+    `Creating ${project.packageManager}${hasWorkspaces ? "workspaces" : ""}`
+  );
   const packageJson = getPackageJson({ workspaceRoot: project.paths.root });
   logger.rootHeader();
-  if (project.packageManager !== "yarn") {
-    logger.rootStep(
-      `adding "workspaces" field to ${project.name} root "package.json"`
-    );
-    packageJson.workspaces = project.workspaceData.globs;
-  }
+
+  // package manager
   logger.rootStep(
-    `adding "packageManager" field to ${project.name} root "package.json"`
+    `adding "packageManager" field to ${path.relative(
+      project.paths.root,
+      project.paths.packageJson
+    )}`
   );
   packageJson.packageManager = `${to.name}@${to.version}`;
 
-  if (!options?.dry) {
-    fs.writeJSONSync(project.paths.packageJson, packageJson, { spaces: 2 });
-  }
+  if (hasWorkspaces) {
+    // workspaces field
+    logger.rootStep(
+      `adding "workspaces" field to ${path.relative(
+        project.paths.root,
+        project.paths.packageJson
+      )}`
+    );
+    packageJson.workspaces = project.workspaceData.globs;
 
-  // root dependencies
-  updateDependencies({
-    workspace: { name: "root", paths: project.paths },
-    project,
-    to,
-    logger,
-    options,
-  });
+    // write package.json here instead of deferring to avoid negating the changes made by updateDependencies
```

Comment:
Is it possible to gracefully feed forward `packageJson`?

Reply:
We could, this gets a little annoying because we're dealing with _multiple_ `package.json`s but here we only want to feed through root (which may not even need to be updated by `updateDependencies` if there are no root internal deps). But I'll see if I can come up with a better API here to avoid the possible double read / write - it annoys me as well. f